flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/strategy/LazyFromSourcesSchedulingStrategy.java

```diff
@@ -142,7 +142,7 @@ private void allocateSlotsAndDeployExecutionVertices(
 
 		schedulerOperations.allocateSlotsAndDeploy(
 			IterableUtils.toStream(vertices)
-				.filter(isInputConstraintSatisfied().and(IS_IN_CREATED_EXECUTION_STATE))
```

Comment:
It seems we still use Java Stream here

Reply:
ok I notice that it is in a separated commit for an on-the-fly improvement. Make sense.